pkg/plugin/velero/item_snapshotter/v1/item_snapshotter.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"time"
+
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+type AlsoHandlesInput struct {
+	ItemToSnapshot runtime.Unstructured
+	Backup         *api.Backup
+}
+
+type SnapshotItemInput struct {
+	Item   runtime.Unstructured
+	Params map[string]string
+	Backup *api.Backup
+}
+
+type SnapshotItemOutput struct {
+	UpdatedItem      runtime.Unstructured
+	SnapshotID       string
+	SnapshotMetadata map[string]string			// SnapshotMetadata is information in addition to the SnapshotID that the
+												// plugin wants to store in the backup.  SnapshotMetadata will be passed
+												// back in for DeleteSnapshot and CreateItemFromSnapshot
+	AdditionalItems  []velero.ResourceIdentifier
+	HandledItems     []velero.ResourceIdentifier // Items that were handled by this snapshot that can be ignored
```

Comment:
As for this slice, does it mean when `SnapshotItem` is called, multiple items may be snapshotted actually?
If the answer is yes, then multiple items may be modified?  Then why the `UpdatedItem` is a single item instead of a slice?

Reply:
HandledItems will be excluded from the backup and won't be touched by anything (including BackupItemActions) so there's no need to be able to modify the copy that is stored in the backup (if these need to be kept, they can be serialized and returned as part of the SnapshotMetadata which Velero will store unmodified)